web-app/js/portal/filter/ui/FilterGroupPanel.js

```diff
@@ -281,12 +281,19 @@ Portal.filter.ui.FilterGroupPanel = Ext.extend(Ext.Container, {
     },
 
     _clearFilters: function() {
+        var that = this;
         Ext.each(this.filterPanels, function(panel) {
-            panel.handleRemoveFilter();
+            if (!that._isSpatialConstraint(panel)) {
+                panel.handleRemoveFilter();
+            }
         });
         this._updateLayerFilters();
     },
 
+    _isSpatialConstraint: function(panel) {
```

Comment:
Logic is fine. Expect it to become something with an `||` condition as the code progress (or doesn't).